packages/expo-image/src/ExpoImage.web.tsx

```diff
@@ -201,27 +209,56 @@ function getFetchPriorityFromImagePriority(priority: ImageNativeProps['priority'
   return priority && ['low', 'high'].includes(priority) ? priority : 'auto';
 }
 
-export default function ExpoImage({
+function Image({
   source,
-  placeholder,
-  contentFit,
+  events,
   contentPosition,
-  onLoad,
-  transition,
-  onError,
-  responsivePolicy,
-  onLoadEnd,
+  blurhashContentPosition,
   priority,
-  ...props
-}: ImageNativeProps) {
-  const { aspectRatio, backgroundColor, transform, borderColor, ...style } = props.style ?? {};
-  const [state, handlers] = useImageState(source);
-  const { placeholder: placeholderStyle, image: imageStyle } = useTransition(transition, state);
-
-  const { containerRef, source: selectedSource } = useSourceSelection(source, responsivePolicy);
+  style,
+  blurhashStyle,
+}: {
+  source?: ImageSource | null;
+  events?: {
+    onLoad: (((event: React.SyntheticEvent<HTMLImageElement, Event>) => void) | undefined)[];
+    onError: ((({ source }: { source: ImageSource | null }) => void) | undefined)[];
+  };
+  contentPosition?: ImageContentPositionObject;
+  blurhashContentPosition?: ImageContentPositionObject;
+  priority?: string | null;
+  style: React.CSSProperties;
+  blurhashStyle?: React.CSSProperties;
+}) {
+  const isBlurhash = isBlurhashString(source?.uri || '');
+  const blurhashUri = useBlurhash(isBlurhash ? source?.uri : null, source?.width, source?.height);
+  const objectPosition = getObjectPositionFromContentPositionObject(
+    isBlurhash ? blurhashContentPosition : contentPosition
+  );
+  const uri = isBlurhash ? blurhashUri : source?.uri;
+  return (
+    <img
+      src={uri || undefined}
```

Comment:
@tsapeta 
I'm not sure how to avoid this explicit undefined:

`null` causes a typing error
`default`, `initial`, `` all show an error icon
`...{uri ? {src: uri} : {}}` seems like a lot of hard to read code for exactly the same outcome